torch_xla/experimental/xla_sharding.py

```diff
@@ -407,9 +416,15 @@ def mark_sharding(t: Union[torch.Tensor, XLAShardedTensor], mesh: Mesh,
     torch_xla._XLAC._xla_mark_sharding(t.global_tensor, tile_assignment,
                                        group_assignment, replication_groups,
                                        int(sharding_type))
+    while tensor_expand:
```

Comment:
Can we again calculate the dim and do one squeeze?

Also, you can make a in-function helper so you can reuse it here and later.